python_modules/dagster/dagster/_core/definitions/partition.py

```diff
@@ -1211,3 +1211,117 @@ def __repr__(self) -> str:
     @classmethod
     def empty_subset(cls, partitions_def: PartitionsDefinition[T_str]) -> "PartitionsSubset[T_str]":
         return cls(partitions_def=partitions_def)
+
+
+class AllPartitionsSubset(
+    PartitionsSubset,
+    NamedTuple(
+        "_AllPartitionsSubset",
+        [
+            ("partitions_def", PartitionsDefinition),
+            ("dynamic_partitions_store", Optional[DynamicPartitionsStore]),
+        ],
+    ),
+):
+    """This is an in-memory (i.e. not serializable) convenience class that represents all partitions
+    of a given PartitionsDefinition, allowing set operations to be taken without having to load
+    all partition keys immediately.
+    """
+
+    def __new__(
+        cls,
+        partitions_def: PartitionsDefinition,
+        dynamic_partitions_store: Optional[DynamicPartitionsStore],
+    ):
+        return super().__new__(
+            cls, partitions_def=partitions_def, dynamic_partitions_store=dynamic_partitions_store
+        )
+
+    @property
+    def partitions_def(self) -> PartitionsDefinition:
```

Comment:
If you switch the order of the superclasses in the class declaration, this might not be required